### PR TITLE
♿️ Settings notifications multiple toggles

### PR DESCRIPTION
**Pull request by cdolm92**
# 📲 What

Provides user with more context as to what notifications they’re toggling. 

# 🤔 Why

The toggles are also not explicitly stating their purpose since these are custom components we’re using.

# 🛠 How

Disabling the title helped us structure the information in a better manner for visually impaired users.

# 👀 See

Trello, screenshots, external resources?

| Before 🐛 | After 🦋 |
| --- | --- |
|unnamed 1| unnamed|

# ♿️ Accessibility 

Navigate to `Settings > Notifications`. Then turn VoiceOver ON.

- [x] Tap on the email notification of each notification. 
          If selected: VoiceOver should read: `[Notification Name], email notification on - Double tap to toggle setting`
   If deselected: VoiceOver should read: `[Notification Name], email notification off - Double tap to toggle setting`
- [x] Toggle email notification:
         If toggled on VoiceOver should read: `[Notification Name], email notification on`
         If toggled off VoiceOver should read: `[Notification Name], email notification off`
- [x] Tap on the push notification of each notification. 
          If selected: VoiceOver should read: `[Notification Name], push notification on - Double tap to toggle setting`
   If deselected: VoiceOver should read: `[Notification Name], push notification off - Double tap to toggle setting`
- [x] Toggle push notification:
         If toggled on VoiceOver should read: `[Notification Name], push notification on`
         If toggled off VoiceOver should read: `[Notification Name], push notification off`

### CODE REVIEW

File: Kickstarter-iOS/Views/Cells/SettingsNotificationCell.swift [hunk 1]
```diff
@@ -71,10 +74,11 @@ final class SettingsNotificationCell: UITableViewCell, NibLoading, ValueCell {
       |> UIButton.lens.image(for: .selected) .~ Library.image(named: "email-icon",
                                                       tintColor: .ksr_green_700,
                                                       inBundle: Bundle.framework)
-      |> UIButton.lens.accessibilityLabel %~ { _ in Strings.Email_notifications() }
 
     _ = self.pushNotificationsButton
       |> notificationButtonStyle
+      |> \.accessibilityTraits .~ .none
+      |> UIButton.lens.accessibilityHint %~ { _ in Strings.Double_tap_to_toggle_setting() }
```
Comment (Scollaco) [inline Kickstarter-iOS/Views/Cells/SettingsNotificationCell.swift]: Since the `accessibilityHint` string is the same for both push and email buttons, I would consider moving this line to the `notificationButtonStyle`, avoiding to repeat ourselves (The same would apply for `\.accessibilityTraits .~ .none`).

Comment (justinswart) [reply Kickstarter-iOS/Views/Cells/SettingsNotificationCell.swift]: Do we include strings in any of our other style functions?

Comment (Scollaco) [reply Kickstarter-iOS/Views/Cells/SettingsNotificationCell.swift]: The `settingsLogoutButtonStyle` includes strings

Comment (cdolm92) [reply Kickstarter-iOS/Views/Cells/SettingsNotificationCell.swift]: @justinswart We do!

File: Kickstarter-iOS/Locales/ja.lproj/Localizable.strings [hunk 2]
```diff
@@ -371,7 +376,7 @@
 "Rewards_count_rewards_colon.one" = "%{rewards_count} 種類のリワード";
 "Rewards_count_rewards_colon.two" = "%{rewards_count} 種類のリワード";
 "Rewards_count_rewards_colon.zero" = "%{rewards_count} 種類のリワード";
-"Save" = "救う";
+"Save" = "保存";
```
Comment (dusi) [inline Kickstarter-iOS/Locales/ja.lproj/Localizable.strings]: 🤔

File: Kickstarter-iOS/Views/Cells/SettingsNotificationCell.swift [hunk 3]
```diff
@@ -38,6 +38,7 @@ final class SettingsNotificationCell: UITableViewCell, NibLoading, ValueCell {
 
     _ = titleLabel
       |> UILabel.lens.text .~ cellValue.cellType.title
+      |> \.accessibilityElementsHidden .~ cellValue.cellType.accessibilityElementsHidden
```
Comment (dusi) [inline Kickstarter-iOS/Views/Cells/SettingsNotificationCell.swift]: Is there a reason we're using `accessibilityElementsHidden` and not `isAccessibilityElement`? I'm not sure that `titleLabel` can have any subviews?

File: Kickstarter-iOS/Views/Cells/SettingsNotificationCell.swift [hunk 4]
```diff
@@ -92,9 +91,12 @@ final class SettingsNotificationCell: UITableViewCell, NibLoading, ValueCell {
 
     self.emailNotificationsButton.rac.selected = viewModel.outputs.emailNotificationsEnabled
     self.emailNotificationsButton.rac.hidden = viewModel.outputs.emailNotificationButtonIsHidden
+    self.emailNotificationsButton.rac.accessibilityLabel =
+      viewModel.outputs.emailNotificationAccessibilityLabel
```
Comment (dusi) [inline Kickstarter-iOS/Views/Cells/SettingsNotificationCell.swift]: Can we change the names of these 2 signals?

So instead of `emailNotificationAccessibilityLabel` do `emailNotificationsButtonAccessibilityLabel` and also instead of `pushNotificationAccessibilityLabel` do `pushNotificationsButtonAccessibilityLabel`.

This should follow the pattern of `component+property`.

File: Library/Strings.swift [hunk 5]
```diff
@@ -20182,7 +20267,7 @@ projets enregistrés"
    - **de**: "%{created_count} erstellt"
    - **es**: "%{created_count} creados"
    - **fr**: "%{created_count} projets créés"
-   - **ja**: "%{created_count} プロジェクト"
+   - **ja**: "%{created_count}プロジェクト"
```
Comment (dusi) [inline Library/Strings.swift]: I wonder if we should get rid of these strings that are unrelated to this PR and maybe review them before they slip to our code base? It looks weird that `プ` would replace a space character to me (even though I don't speak Japanese).

What do you think Christella?

File: Kickstarter-iOS/Views/Cells/SettingsNotificationCell.swift (hunk 4 above)
Comment (cdolm92) [reply Kickstarter-iOS/Views/Cells/SettingsNotificationCell.swift]: Because these buttons behave more like switches I was not sure that referring to it as a button . was accurate. Let me know what you think.

File: Library/Strings.swift (hunk 5 above)
Comment (cdolm92) [reply Library/Strings.swift]: Ideally but maybe we should have a team discussion on handling strings? I think we can change this approach later.

File: Kickstarter-iOS/Views/Cells/SettingsNotificationCell.swift (hunk 4 above)
Comment (dusi) [reply Kickstarter-iOS/Views/Cells/SettingsNotificationCell.swift]: I realize we're not really using them as buttons, but technically the component (`UIButton`) and the name of the component (`emailNotificationsButton`) refer to a button so we should probably stay consistent. The behaviour should be considered a side effect IMO

File: Library/Strings.swift (hunk 5 above)
Comment (dusi) [reply Library/Strings.swift]: Agree, we should bring this up with Nneka & Peat (either parking lot or slack)?

File: Kickstarter-iOS/Views/Cells/SettingsNotificationCell.swift [hunk 6]
```diff
@@ -83,7 +83,6 @@ final class SettingsNotificationCell: UITableViewCell, NibLoading, ValueCell {
       |> UIButton.lens.image(for: .selected) .~ Library.image(named: "mobile-icon",
                                                       tintColor: .ksr_green_700,
                                                       inBundle: Bundle.framework)
-      |> UIButton.lens.accessibilityLabel %~ { _ in Strings.Push_notifications() }
```
Comment (dusi) [inline Kickstarter-iOS/Views/Cells/SettingsNotificationCell.swift]: Noticed that there was previously an empty line here, can we possibly remove it?

File: Kickstarter-iOS/Views/Cells/SettingsNotificationCell.swift (hunk 3 above)
Comment (cdolm92) [reply Kickstarter-iOS/Views/Cells/SettingsNotificationCell.swift]: When `isAccessibilityElement` is set to `false` for the notification cells w/  email/push icons and `true` for `Manage project notifications` button, VoiceOver as expected, would avoid focusing on the notification cell titles, however, for `Manage project notifications` button  VoiceOver was reading the button title and number of notifications separately. If the user tapped over the cell VoiceOver was only reading the number of notifications, the user would have to tap over the title for it to read.  With `accessibilityElementsHidden` VoiceOver avoids focusing on reading the titles of the notification cells w/ email/push icons while still reading the `Manage Project notifications` title and number of notifications together.

Comment (dusi) [reply Kickstarter-iOS/Views/Cells/SettingsNotificationCell.swift]: So the only way I was able to have this work properly was to use your original approach + fix the order of the buttons like so:

```
func configureWith(value cellValue: SettingsNotificationCellValue) {
    self.notificationType = cellValue.cellType

    viewModel.inputs.configure(with: cellValue)

    let accessibilityElementsHidden = cellValue.cellType.accessibilityElementsHidden

    _ = self
      |> \.accessibilityTraits .~ cellValue.cellType.accessibilityTraits

    _ = self.stackView
      |> \.accessibilityElements .~ (
        accessibilityElementsHidden ? [self.emailNotificationsButton, self.pushNotificationsButton] : nil
    )

    _ = self.titleLabel
      |> UILabel.lens.text .~ cellValue.cellType.title
      |> \.accessibilityElementsHidden .~ accessibilityElementsHidden

    _ = self.arrowImageView
      |> UIImageView.lens.isHidden .~ cellValue.cellType.shouldHideArrowView
      |> UIImageView.lens.tintColor .~ .ksr_dark_grey_400

    _ = self.projectCountLabel
      |> UILabel.lens.isHidden .~ cellValue.cellType.projectCountLabelHidden
      |> \.accessibilityElementsHidden .~ accessibilityElementsHidden
  }
```

please notice that:
* I've added `self.` to reference `IBOutlet`s consistently
* Also set `accessibilityElementsHidden` on the `projectCountLabel` (it worked without it but might be confusing as to why since we don't won't either of the labels to be accessible and let the cell take over instead)

File: Library/SettingsNotificationCellType.swift [hunk 7]
```diff
@@ -81,6 +81,15 @@ public enum SettingsNotificationCellType {
     }
   }
 
+  public var accessibilityElementsHidden: Bool {
+    switch self {
+    case .projectNotifications, .emailFrequency:
```
Comment (dusi) [inline Library/SettingsNotificationCellType.swift]: Do we need `.emailFrequency` here at all? The cell we're using for that is `SettingsNotificationPickerCell` which doesn't seem to use this value, yet it still works.
